code/modules/clothing/suits/reactive_armour.dm

```diff
@@ -0,0 +1,235 @@
+/obj/item/reactive_armour_shell
+	name = "reactive armour shell"
+	desc = "An experimental suit of armour, awaiting installation of an anomaly core."
+	icon_state = "reactiveoff"
+	icon = 'icons/obj/clothing/suits.dmi'
+	w_class = WEIGHT_CLASS_BULKY
+
+/obj/item/reactive_armour_shell/attackby(obj/item/I, mob/user, params)
+	..()
+	var/static/list/anomaly_armour_types = list(
+		/obj/effect/anomaly/grav	                = /obj/item/clothing/suit/armor/reactive/repulse,
+		/obj/effect/anomaly/flux 	           		= /obj/item/clothing/suit/armor/reactive/tesla,
+		/obj/effect/anomaly/bluespace 	            = /obj/item/clothing/suit/armor/reactive/teleport,
+		/obj/effect/anomaly/pyro	  			    = /obj/item/clothing/suit/armor/reactive/fire)
+
+	if(istype(I, /obj/item/assembly/signaler/anomaly))
+		var/obj/item/assembly/signaler/anomaly/A = I
+		var/armour_path = anomaly_armour_types[A.anomaly_type]
+		if(!armour_path)
+			armour_path = /obj/item/clothing/suit/armor/reactive/stealth //Lets not cheat the player if an anomaly type doesnt have its own armour coded
+		to_chat(user, "You insert [A] into the chest plate, and the armour gently hums to life.")
+		new armour_path(get_turf(src))
+		qdel(src)
+		qdel(A)
+
+//Reactive armor
+/obj/item/clothing/suit/armor/reactive
```

Comment:
Don't forget to make a subtype of this that has some special sprite/name/what have you, that is the steal object, so people can't just make a shell+use an anomaly to complete their objective (granted, the chances of that happening is rare, but the precedent has generally been you have to go for a one of a kind item).

Reply:
If someone goes that far they should get the obj tbh.

Reply:
Getting a bluespace anomaly is rare enough that I dont think a subtype is warranted